roles/provision-crc-apb/tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+---
+- name: Validate CRC JSON config
+  set_fact: svc_json_config={{ SERVICE_CONFIG | from_json }}
+  register: crc_json_config_valid
+  ignore_errors: True
+
+- fail:
+    msg: |
+      Failed to validate provided JSON configuration
+      CRC with invalid JSON config can not be created
+  when: crc_json_config_valid|failed
+
+- name: Creating CRC {{ crc_mobile_config_name }} secrets
+  k8s_v1_secret:
+    name: '{{ crc_mobile_config_name }}'
+    namespace: '{{ namespace }}'
+    labels:
+      name: '{{ SERVICE_NAME }}'
+      mobile: '{{ crc_mobile_svc_label }}'
+      serviceName: '{{ crc_mobile_svc_name }}'
+    string_data:
+      type: '{{ crc_mobile_svc_type }}'
+      name: '{{ SERVICE_NAME }}'
+      uri: '{{ SERVICE_URI }}'
+
+- name: Creating CRC {{ crc_mobile_config_name }} configmap
+  k8s_v1_config_map:
+    name: '{{ crc_mobile_config_name }}'
+    namespace: '{{ namespace }}'
+    annotations:
+      configType: '{{ crc_mobile_config_type }}'
+    labels:
+      name: '{{ SERVICE_NAME }}'
+      mobile: '{{ crc_mobile_svc_label }}'
+      serviceName: '{{ crc_mobile_svc_name }}'
+    data:
+      type: '{{ crc_mobile_svc_type }}'
+      name: '{{ SERVICE_NAME }}'
+      uri: '{{ SERVICE_URI }}'
+      config: '{{ svc_json_config | to_nice_json }}'
+
+- name: Get the name of the service instance
```

Comment:
we don't need to do this as, we are doing work to remove the need for this labeling as part of this sprint